djangocms_text_ckeditor/static/djangocms_text_ckeditor/ckeditor_plugins/cmsplugins/plugin.js

```diff
@@ -41,36 +77,73 @@ $(document).ready(function () {
 						anchors.bind('click', function (e) {
 							e.preventDefault();
 
-							that.addPlugin($(this), panel);
+							that.addPlugin($(this).rel, panel);
 						});
 				}
 			});
+		},
 
-			// handle edit event via context menu
-			if(this.editor.contextMenu) {
-				this.setupContextMenu();
-				this.editor.addCommand('cmspluginsEdit', {
-					exec: function () {
-						var selection = that.editor.getSelection();
-						var element = selection.getSelectedElement() || selection.getCommonAncestor().getAscendant('a', true);
-						that.editPlugin(element);
-					}
-				});
+		_defaultPluginIcons: [
+			{
+				path: 'icons/cmspluginsMap.png',
+				match: function(item) { return item.type.match('Map'); }
+			},
+			{
+				path: 'icons/cmspluginsVideo.png',
+				match: function(item) { return item.type.match('Video'); }
+			},
+			{
+				path: 'icons/cmspluginsAudio.png',
+				match: function(item) { return item.type.match('Audio'); }
+			},
+			{
+				path: 'icons/cmspluginsImage.png',
+				match: function(item) { return item.type.match('Image'); }
+			},
+			{
+				path: 'icons/cmspluginsFile.png',
+				match: function(item) { return item.type.match('File'); }
+			},
+			{
+				path: 'icons/cmspluginsLink.png',
+				match: function(item) { return item.type.match('Link'); }
+			}
+		],
+
+		_getPluginIconPath: function (item) {
+			if (item.icon_path)
+				return item.icon_path;
+
+			// try to guess 
+			var icon = null;
+			for(i = 0; i < this._defaultPluginIcons.length; i++) {
+				icon = this._defaultPluginIcons[i];
+				if (icon.match(item))
+					return this.path + icon.path;
 			}
 
-			// hande event via doubleclick
-			// handle edit event on double click
-			this.editor.on('doubleclick', function(event) {
-				var selection = that.editor.getSelection();
-				var element = selection.getSelectedElement() || selection.getCommonAncestor().getAscendant('a', true);
-				if(element && element.getAttribute('id').indexOf('plugin_obj_') === 0 && !element.isReadOnly()) {
-					event.data.dialog = '';
-					that.editPlugin(element);
-				}
-			});
+			// fallback
+			return this.path + 'icons/cmsplugins.png';
+		},
 
-			// setup CKEDITOR.htmlDataProcessor
-			this.setupDataProcessor();
+		_initButtons: function () {
+			var that = this;
+			$.each(this.options.plugins, function (i, group) {
```

Comment:
its very dangerous to use `i` and than `ii` give them meaningful names like pluginIndex or groupIndex so we can distinguish the both of them while reading the code